src/io/cached_input_split.h

```diff
@@ -88,7 +88,7 @@ class CachedInputSplit : public InputSplit {
     LOG(FATAL) << "ResetPartition is not supported in CachedInputSplit";
   }
   virtual void HintChunkSize(size_t chunk_size) {
-    buffer_size_ = std::max(chunk_size / sizeof(size_t), buffer_size_);
+    buffer_size_ = std::max(chunk_size / sizeof(uint32_t), buffer_size_);
```

Comment:
Even though recordio is aligned to 4 bytes, reader was reading to chunk's data vector which was size_t (so 8 bytes aligned). This is problematic, because there is no guarantee that recordio entry is aligned to 8 bytes. That's why I changed chunk's internal data vector to uint32_t